api/server/routes_create_update.go

```diff
@@ -66,10 +89,16 @@ func (s *Server) changeRoute(ctx context.Context, wroute *models.RouteWrapper) e
 	return nil
 }
 
-// ensureApp will only execute if it is on put
-func (s *Server) ensureRoute(ctx context.Context, method string, wroute *models.RouteWrapper) (routeResponse, error) {
+// ensureApp will only execute if it is everything except PATCH
+func (s *Server) ensureRoute(ctx context.Context, appID, method string, wroute *models.RouteWrapper) (routeResponse, error) {
 	bad := new(routeResponse)
 
+	app, err := s.Datastore().GetAppByID(ctx, appID)
```

Comment:
we don't need to do this? just stick the id onto `wroute.Route.AppID = appID` - if it doesn't find it, we'll find out soon enough, and we got the appID from the app name in the request already so it should be safe to use?

Reply:
from `ensureApp`*